spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -370,14 +371,56 @@ public void alterNamespace(String[] namespace, NamespaceChange... changes) throw
 
   @Override
   public boolean dropNamespace(String[] namespace) throws NoSuchNamespaceException {
+    // Spark assumes that catalogs CASCADE by default. So if we have a namespace catalog
+    /// we need to attempt to recursively delete.
     if (asNamespaceCatalog != null) {
+      Namespace asNamespace = Namespace.of(namespace);
+      boolean exists = namespaceExists(namespace);
```

Comment:
I think this is redundant. If the namespace doesn't exist then the call to `listNamespaces` will throw Spark's `NoSuchNamespaceException` and this can exit early. Otherwise, the namespace exists in some form. The only reason to call `namespaceExists` is if there is a custom implementation of that method (rather than loading metadata and catching `NoSuchNamespace`) but there isn't, so we may as well not make too many calls.